src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect } from 'react';
+import { Range } from 'react-range';
+
+import useTheme from '../../hooks/useTheme';
+import { TestProps } from '../../utils/types';
+import TextField from '../TextField';
+import SliderMark from './components/SliderMark';
+import SliderThumb from './components/SliderThumb';
+import SliderTrack from './components/SliderTrack';
+import { Container, InputContainer, InputsContainer } from './Slider.style';
+
+interface Props {
```

Comment:
If `Selector` only has 1 value and `Range` only has 2 values, the props could be split as 

```ts
type CommonProps = {
  disabled?: boolean;
  hasIncrements?: boolean;
  onChange: (values: number[]) => void;
  onFinalChange?: (values: number[]) => void;
}

type SelectorProps = {
  values: [number];
  isSelector: true;
  initialValues?: [number];
};

type RangeProps = {
  values: [number, number];
  isSelector: false;
  initialValues?: [number, number];
};

type Props = CommonProps & (SelectorProps | RangeProps);
```